patroni/ha.py

```diff
@@ -1938,23 +1933,28 @@ def get_remote_member(self, member: Union[Leader, Member, None] = None) -> Remot
         name = member.name if member else 'remote_member:{}'.format(uuid.uuid1())
         return RemoteMember(name, data)
 
-    def get_failover_candidates(self, check_sync: bool = False) -> List[Member]:
-        """Return list of candidates for either manual or automatic failover.
+    def get_failover_candidates(self, check_synchronous: bool, exclude_candidate: bool) -> List[Member]:
+        """Return a list of candidates for either manual or automatic failover.
 
-        Mainly used to later be passed to ``Ha.is_failover_possible()``.
+        Exclude non-sync members when in synchronous mode, the current node (its checks are always performed earlier)
+        and the candidate if required. If failover candidate exclusion is not requested and a candidate is specified
+        in the /failover key, return the candidate only.
+        The result is further evaluated in the caller :func:`Ha.is_failover_possible` to check if any member is actually
+        healthy enough and is allowed to poromote.
 
-        :param check_sync: if ``True``, also check against the sync key members
+        :param check_synchronous: if ``True``, also check against the sync key members.
+        :param exclude_candidate: if ``True``, exclude :attr:`failover.candidate` from the candidates.
 
-        :returns: a list of ``Member`` ojects or an empty list if there is no candidate available
+        :returns: a list of :class:`Member` ojects or an empty list if there is no candidate available.
         """
-        failover = self.cluster.failover
-        if check_sync:
+        def is_eligible(node_name: str, check_synchronous: bool, exclude_candidate: bool) -> bool:
```

Comment:
We better pass `Member` object, so we can add additional checks for api_url and that nofailover isn't set.
check_synchronous and check_synchronous are not needed here.
```suggestion
        def is_eligible(node: Member) -> bool:
```